examples-gallery/plot_ball_rolling_on_spinning_disc.py

```diff
@@ -0,0 +1,620 @@
+ # %%
+"""
+Ball Rolling on Spinning Disc
+=============================
+
+A uniform, solid ball with radius r and mass :math:`m_b` is rolling on a
+horizontal spinning disc without slipping.
+The disc starts at rest and speeds up like this
+:math:`u_{3}(t) = \\Omega \\cdot (1 - e^{(-\\alpha \\cdot t)})`
+where :math:`\\alpha` > 0 is a measure of the acceleration and :math:`\\Omega`
+is the final rotational speed.
+A torque is applied to the ball, and the goal is to get it to the
+center of the disc.
+
+An observer, a particle of mass :math:`m_o`, is attached
+to the surface the ball.
+
+**Constants**
+
+- :math:`m_b`: mass of the ball [kg]
+- r : radius of the ball [m]
+- :math:`m_o` : mass of the observer [kg]
+- :math:`\\Omega`: final rotational speed of the disc
+  around the vertical axis [rad/sec]
+- :math:`\\alpha`: measure of the acceleration of the disc [1/sec]
+
+**States**
+
+- :math:`q_1, q_2, q_3`: generalized coordinates of the ball
+    w.r.t the disc [rad]
+- :math:`u_1, u_2, u_3`: generalized angual velocities of the ball [rad/sec]
+
+**Specifieds**
+
+- :math:`t_1, t_2, t_3`: Torque applied to the ball [N * m]
+
+**Additional parameters**
+
+- ``N``: inertial frame
+- ``A1``: frame fixed to the ball
+- ``A2``: frame fixed to the disc
+- ``O``: point fixed in N
+- ``CP``: contact point of ball with disc
+- ``Dmc``: center of the ball
+- :math:`m_{Dmc}`: position of observer
+- ``h``: intervall which opty should minimize.
```

Comment:
You display some of these as LaTeX math and some as code. Best to be consistent and pick one or the other.